qiskit/circuit/controlledgate.py

```diff
@@ -119,21 +124,21 @@ def definition(self):
         return open_rules + self._definition + open_rules
 
     @definition.setter
-    def definition(self, excited_def):
+    def definition(self, excited_def: List):
         """Set controlled gate definition with closed controls."""
         super(Gate, self.__class__).definition.fset(self, excited_def)
 
     @property
-    def ctrl_state(self):
+    def ctrl_state(self) -> int:
         """Return the control state of the gate as a decimal integer."""
         return self._ctrl_state
 
     @ctrl_state.setter
-    def ctrl_state(self, ctrl_state):
+    def ctrl_state(self, ctrl_state: Union[int, str, None]):
```

Comment:
```suggestion
    def ctrl_state(self, ctrl_state: Optional[int, str]):
```

Reply:
Since the setter is required to take a value I didn't make it Optional.